api/api.go

```diff
@@ -55,92 +43,82 @@ func handleError(w http.ResponseWriter, err error) {
+					handleError(w, authError("invalid token"))
+					return
+				}
 			}
 
 			next.ServeHTTP(w, r)
 		})
 	}
 }
 
-func NewRouter(repo *repository.Repository, token string) *mux.Router {
+func SetupRoutes(server *http.ServeMux, repo *repository.Repository, token string) {
 	lstore = repo.Store()
 	lrepository = repo
 
-	r := mux.NewRouter()
-
-	apiRouter := r.PathPrefix("/api").Subrouter()
-	if token != "" {
-		apiRouter.Use(AuthMiddleware(token))
-	}
-
+	authToken := TokenAuthMiddleware(token)
 	urlSigner := NewSnapshotReaderURLSigner(token)
 
-	readerRouter := r.PathPrefix("/api").Subrouter()
-	if token != "" {
-		readerRouter.Use(urlSigner.VerifyMiddleware)
-	}
-
-	handle := func(path string, handler func(http.ResponseWriter, *http.Request) error) *mux.Route {
-		return apiRouter.Handle(path, Handler(handler))
-	}
-
-	readerHandle := func(path string, handler func(http.ResponseWriter, *http.Request) error) *mux.Route {
-		return readerRouter.Handle(path, Handler(handler))
-	}
-
-	handle("/storage/configuration", storageConfiguration).Methods("GET")
-	handle("/storage/states", storageStates).Methods("GET")
-	handle("/storage/state/{state}", storageState).Methods("GET")
-	handle("/storage/packfiles", storagePackfiles).Methods("GET")
-	handle("/storage/packfile/{packfile}", storagePackfile).Methods("GET")
-
-	handle("/repository/configuration", repositoryConfiguration).Methods("GET")
-	handle("/repository/snapshots", repositorySnapshots).Methods("GET")
-	handle("/repository/states", repositoryStates).Methods("GET")
-	handle("/repository/state/{state}", repositoryState).Methods("GET")
-	handle("/repository/packfiles", repositoryPackfiles).Methods("GET")
-	handle("/repository/packfile/{packfile}", repositoryPackfile).Methods("GET")
-
-	handle("/snapshot/{snapshot}", snapshotHeader).Methods("GET")
-	readerHandle("/snapshot/reader/{snapshot}:{path:.+}", snapshotReader).Methods("GET")
-	handle("/snapshot/reader-sign-url/{snapshot}:{path:.+}", urlSigner.Sign).Methods("POST")
-	handle("/snapshot/search/{snapshot}:{path:.+}", snapshotSearch).Methods("GET")
-
-	handle("/snapshot/vfs/{snapshot}:/", snapshotVFSBrowse).Methods("GET")
-	handle("/snapshot/vfs/{snapshot}:{path:.+}/", snapshotVFSBrowse).Methods("GET")
-	handle("/snapshot/vfs/{snapshot}:{path:.+}", snapshotVFSBrowse).Methods("GET")
-
-	handle("/snapshot/vfs/children/{snapshot}:/", snapshotVFSChildren).Methods("GET")
-	handle("/snapshot/vfs/children/{snapshot}:{path:.+}/", snapshotVFSChildren).Methods("GET")
-	handle("/snapshot/vfs/children/{snapshot}:{path:.+}", snapshotVFSChildren).Methods("GET")
-
-	handle("/snapshot/vfs/errors/{snapshot}:/", snapshotVFSErrors).Methods("GET")
-	handle("/snapshot/vfs/errors/{snapshot}:{path:.+}/", snapshotVFSErrors).Methods("GET")
-	handle("/snapshot/vfs/errors/{snapshot}:{path:.+}", snapshotVFSErrors).Methods("GET")
-
-	return r
+	server.Handle("/api/{path...}", APIView(func(w http.ResponseWriter, r *http.Request) error {
```

Comment:
micro nit: i won't use the path parameter when we're not going to use it, just `"/api/"` should do it here.  anyway, it looks fine to me, thanks!

Reply:
done